### PR TITLE
Doc and tests for Stdune.String.split

### PR DESCRIPTION
**Pull request by gridbugs**

### CODE REVIEW

File: otherlibs/stdune/src/string.mli [hunk 1]
```diff
@@ -52,7 +52,15 @@ val extract_blank_separated_words : t -> t list
 val lsplit2 : t -> on:char -> (t * t) option
 val lsplit2_exn : t -> on:char -> t * t
 val rsplit2 : t -> on:char -> (t * t) option
+
+(** [split t ~on] returns the list of non-overlapping substrings of
+    [t] between each occurence of [on]. If [t] begins or ends with [on]
+    then an empty string will be present on the "far" side of [on] in the
+    output. If [on] does not appear in [t] then the result is a list
+    containing [t] (even if [t] is the empty string). This function never
+    returns the empty list. *)
```
Comment (Leonidas-from-XIV) [inline otherlibs/stdune/src/string.mli:61]: I think the idea here is that joining a string that was split would return the original string, this it needs to have empty strings on the far ends, so the original locations of the separators can be recovered.

Maybe worth mentioning.

Comment (gridbugs) [reply otherlibs/stdune/src/string.mli:61]: Makes sense. Will update. To be clear I think the way `split` works is very natural. Just wanted to make the doc explicitly state how it behaves around edge cases.

File: otherlibs/stdune/src/string.mli [hunk 2]
```diff
@@ -52,7 +52,20 @@ val extract_blank_separated_words : t -> t list
...
+(** [split t ~on] returns the list of non-overlapping substrings of
+    [t] between each occurence of [on]. If [t] begins or ends with [on]
+    then an empty string will be present on the "far" side of [on] in the
+    output. If [on] does not appear in [t] then the result is a list
+    containing [t] (even if [t] is the empty string). This function never
+    returns the empty list.
+
+    Calling [concat] on the result of [split] with the same delimiter
+    (converted from a [char] to a [string]) will produce its original
+    argument. (The inverse is not true however as [concat] can be
```
[the remark this answers is not in the document]
Comment (gridbugs) [reply otherlibs/stdune/src/string.mli:65]: Ah yeh maybe that would be clearer.

[the remark this answers is not in the document]
Comment (gridbugs) [reply otherlibs/stdune/src/string.mli]: I tried something. Let me know what you think.

**Review by emillon — APPROVED** [review]

**Pull request merged**